app/celery/contact_information_tasks.py

```diff
@@ -24,55 +38,197 @@
 )
 @statsd(namespace='tasks')
 def lookup_contact_info(
-    self,
-    notification_id,
+    self: Task,
+    notification_id: str,
 ):
+    """
+    Celery task to look up contact information (email/phone number) for a given notification.
+    If the feature flag, VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP, is enabled,
+    also check for related communication permissions.
+
+    Args:
+        self (Task): The Celery task instance.
+        notification_id (str): The ID of the notification for which to look up contact information.
+
+    Raises:
+        AutoRetryException: If a retryable exception occurs during the lookup process.
+        NotificationTechnicalFailureException: If the maximum retries have been exceeded.
+        NotificationPermanentFailureException: If the exception indicates a permanent failure.
+        Exception: If an unhandled exception occurs.
+
+    Returns:
+        None
+    """
     current_app.logger.info('Looking up contact information for notification_id: %s.', notification_id)
 
     notification = get_notification_by_id(notification_id)
     recipient_identifier = notification.recipient_identifiers[IdentifierType.VA_PROFILE_ID.value]
+    should_send = notification.default_send
 
     try:
-        if EMAIL_TYPE == notification.notification_type:
-            recipient = va_profile_client.get_email(recipient_identifier)
-        elif SMS_TYPE == notification.notification_type:
-            recipient = va_profile_client.get_telephone(recipient_identifier)
+        if is_feature_enabled(FeatureFlag.VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP):
+            result = get_profile_result(notification, recipient_identifier)
+            recipient = result.recipient
+            should_send = result.communication_allowed
+            permission_message = result.permission_message
         else:
-            raise NotImplementedError(
-                f'The task lookup_contact_info failed for notification {notification_id}. '
-                f'{notification.notification_type} is not supported'
+            recipient = get_recipient(
+                notification.notification_type,
+                notification_id,
+                recipient_identifier,
             )
-    except (Timeout, VAProfileRetryableException) as e:
-        if can_retry(self.request.retries, self.max_retries, notification_id):
-            current_app.logger.warning('Unable to get contact info for notification id: %s, retrying', notification_id)
+    except Exception as e:
+        handle_lookup_contact_info_exception(self, notification, recipient_identifier, e)
+
+    notification.to = recipient
+    dao_update_notification(notification)
+
+    if not should_send:
+        handle_communication_not_allowed(notification, recipient_identifier, permission_message)
+
+
+def get_recipient(
```

Comment:
Even though this function is in the green column, this is the original functionality that calls the v1 endpoint.

The diff for this file is difficult to parse.  Thanks for your patience